Upstatement/firestore-jest-mock
# Add basic typing declarations

# Description

This PR represents some initial work on exporting type information to users. More pull requests will follow with typings for each of our exported modules.

## Related issues

Related to #20 

## How to test

1. Create a new local project: `npm init`
2. Install the package via this branch: `npm install --save-dev https://github.com/AverageHelper/firestore-jest-mock.git#typescript-shim`
3. Add a line to index.js to require `firestore-jest-mock`:
    ```JavaScript
    const { mockFirebase } = require("firestore-jest-mock");
    ```
4. Hover over `mockFirebase` to ensure that your editor picks up the type declarations:
    ![Screen Shot 2021-03-11 at 10 09 52 AM](https://user-images.githubusercontent.com/1500092/110826054-ed4adb80-8251-11eb-88c7-bf43ba5ec267.png)
